hail/src/main/scala/is/hail/backend/service/Worker.scala

```diff
@@ -177,13 +177,13 @@ object Worker {
     retryTransientErrors {
       write(s"$root/result.$i") { dos =>
         if (result != null) {
-          assert(userError == null)
+          assert(errorWhileExecutingUserCode == null)
```

Comment:
All this horridness can be avoided if we return `Either[Throwable, Array[Byte]]` from `using` above.

Reply:
Ah, great point. I've updated to use Either; much nicer.